scadnano/scadnano.py

```diff
@@ -3863,16 +3853,59 @@ def from_scadnano_json_map(
         deserialized_helices_list = json_map[helices_key]
         num_helices = len(deserialized_helices_list)
 
-        # create Helices
+
+
+        # USED ONLY IN THE CASE WHERE THERE IS MORE THAN ONE HELIX GROUP
+        # Maps group name to pitch, yaw, and helix idx of a helix in that group (the first one found).
+        # The helix idx is kept in case an error message needs to be thrown.
+        group_to_pitch_yaw: Dict[str, Tuple[float, float, int]] = {}
+
+        # USED ONLY IN THE CASE WHEN THERE IS EXACTLY ONE HELIX GROUP
+        # Maps pitch and yaw pairs to list of helix with those pitch and yaw
+        # Because of fuzziness with floating point values, the keys are actually
+        # int(pitch/floating_point_tolerance) and int(yaw/floating_point_tolerance)

```

Comment:
I don't think this is a good way to test close floating point values. I've never seen this trick before (divide by the desired tolerance and then take a floor).

What if two values are `10.000001` and `9.999999` and `tolerance` is `0.01`? Then `10.000001 / 0.01` = `1000.0001`, which floors to `1000`, but `9.999999 / 0.01` = `999.9999`, which floors to `999`.

There would be similar problems with rounding to the nearest integer, e.g., if the two values were `1.004999` and `1.005001`, divided by `0.01` gives `100.4999` and `100.5001`, which would round to `100` and `101`, respectively, with `tolerance` = `0.01`, even though the numbers are within `0.01` of each other.

I think perhaps this logic could work, but you'd want to test whether adjacent integers are keys in the dict, e.g., if you convert a float to integer 1005, then you'd want any float converted to 1004 or 1006 to be considered equivalent. But I'm not sure if there's a concise way of expressing that.

I think the most readable approach, although it would be a bit more time-consuming (but this is small data, and the case where this is needed is rare), is to store the floating point values directly as keys, but then instead of asking `if yaw in pitch_yaw_to_helices`, instead do the linear-time loop over all keys `for existing_yaw in pitch_yaw_to_helices`, asking `if _is_close(yaw, existing_yaw)`.